app/Models/Membership.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Models;
+
+use App\Models\Membership\MembershipPivot;
+use App\User;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Support\Collection;
+
+class Membership extends Model
+{
+    const IDENT_DIVISION = 'DIV';
+    const IDENT_REGION = 'REG';
+    const IDENT_INTERNATIONAL = 'INT';
+    const IDENT_VISITING = 'VIS';
+    const IDENT_TRANSFERING = 'TFR';
+
+    public $timestamps = false;
+
+    protected $casts = [
+        'primary' => 'bool',
+        'can_have_secondaries' => 'bool',
+    ];
+
+    public function scopePrimary(Builder $query): Builder
+    {
+        return $query->where('primary', true);
+    }
+
+    public function scopeSecondary(Builder $query): Builder
+    {
+        return $query->where('primary', false);
+    }
+
+    public function scopeOrderByPriority(Builder $query): Builder
+    {
+        return $query->orderBy('priority');
+    }
+
+    public static function findByIdent(string $ident): ?self
+    {
+        return self::where('identifier', $ident)->first();
+    }
+
+    public static function findPrimaryByVATSIMLocality(string $division, string $region): ?self
+    {
+        return self::primary()->orderByPriority()->get()->first(function ($membership) use ($division, $region) {
+            $divisions = $membership->division_expression;
+            $regions = $membership->region_expression;
+
+            $regionCriteriaMatched = $regions && ($regions->first() == '*' || $regions->contains($region));
```

Comment:
I have no idea what this is doing

Reply:
It takes the expression for matching with a given region from the membership (e.g. for Regional Membership, the user must have the EUR region), and checks it against the user's region. If the Membership is setup with a wildcard, it will return true and as such the user's region qualifies them for this membership type. If not a wildcard, it searches the collection to find a match.